src/evict/evict_page.c

```diff
@@ -661,9 +663,25 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
             LF_SET(WT_REC_SCRUB);
     }
 
+    saved_pinned_id = WT_SESSION_TXN_SHARED(session)->pinned_id;
+
+    /* Acquire a snapshot if coming through eviction thread route. */
+    if (F_ISSET(session, WT_SESSION_INTERNAL)) {
+        __wt_txn_get_snapshot(session);
+        release_snapshot = true;
+    }
+
     /* Reconcile the page. */
     ret = __wt_reconcile(session, ref, NULL, flags);
 
+    /*
+     * If we got a snapshot in order to write pages, and there was no snapshot active when we
+     * started, release it.
+     */
+    if (release_snapshot && session->txn->isolation == WT_ISO_READ_COMMITTED &&
```

Comment:
we can initialize release_snapshot to false and just based on it we can release the snapshot right? why do we need to check the isolation level ?

Reply:
Fixed.